### PR TITLE
Automated text-to-image solution for training fine-tuned Stable Diffusion XL 1.0 model with Kohya SS

### PR DESCRIPTION
**Pull request by azograby**
*Issue #, if available:*

*Description of changes:* Initial solution for text-to-image fine-tuning SDXL

*Testing done:* Ran through end to end test in own environment, from Cloudformation to Inference. Description of solution and demo located on Broadcast: https://broadcast.amazon.com/videos/942863.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `black-nb -l 100 {path}/{notebook-name}.ipynb`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: use-cases/text-to-image-fine-tuning/template.yml [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+                  - ecs:UpdateService
+                  - ecr:BatchGetImage
+                  - ecr:GetDownloadUrlForLayer
+                Resource: '*'
+              - Effect: Allow
+                Action: 
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource:
```
[the remark this answers is not in the document]
Comment (azograby) [reply use-cases/text-to-image-fine-tuning/template.yml:304]: Added dynamic partition

File: use-cases/text-to-image-fine-tuning/template.yml [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action: 
+                  - codebuild:CreateReportGroup
+                  - codebuild:CreateReport
+                  - codebuild:UpdateReport
+                  - codebuild:BatchPutTestCases
+                  - codebuild:BatchPutCodeCoverages
+                Resource: !Sub 'arn:aws:codebuild:${AWS::Region}:${AWS::AccountId}:report-group/${TrainingContainerBuildProjectName}*'
```
[the remark this answers is not in the document]
Comment (azograby) [reply use-cases/text-to-image-fine-tuning/template.yml]: Added dynamic partition

File: use-cases/text-to-image-fine-tuning/template.yml [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+                DefaultValue: !Join
+                  - ''
+                  - - 's3://'
+                    - !Ref TrainingS3Bucket
+                    - '/'
+                    - '0001-dataset'
+              - Name: OutputS3ModelLocation
+                Type: String
+                Description: The S3 location where the model will be output to
+                DefaultValue: !Join
```
[the remark this answers is not in the document]
Comment (azograby) [reply use-cases/text-to-image-fine-tuning/template.yml]: Changed to Sub

File: use-cases/text-to-image-fine-tuning/template.yml [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+        MaxParallelExecutionSteps: 1 # max num of steps that can be executed in parallel
+      PipelineDefinition: 
+        PipelineDefinitionBody:
+          Fn::ToJsonString: 
+            Version: '2020-12-01'
+            Parameters: 
+              - Name: InputS3DatasetLocation
+                Type: String
+                Description: S3 URI of the training dataset
+                DefaultValue: !Join
```
[the remark this answers is not in the document]
Comment (azograby) [reply use-cases/text-to-image-fine-tuning/template.yml]: Changed to Sub

**Review by zhaoqizqwang — APPROVED** [review]

**Pull request merged**